packages/devtools_app/lib/src/debugger/flutter/debugger_controller.dart

```diff
@@ -312,6 +333,92 @@ class DebuggerController extends DisposableController
     appendStdio(text);
   }
 
+  Future<List<ScriptRef>> _retrieveSortScripts(IsolateRef ref) async {
+    final scriptList = await _service.getScripts(isolateRef.id);
+    // We filter out non-unique ScriptRefs here (dart-lang/sdk/issues/41661).
+    final scriptRefs = Set.of(scriptList.scripts).toList();
+    scriptRefs.sort((a, b) {
+      // We sort uppercase so that items like dart:foo sort before items like
+      // dart:_foo.
+      return a.uri.toUpperCase().compareTo(b.uri.toUpperCase());
+    });
+    return scriptRefs;
+  }
+
+  void _updateAfterIsolateReload(Event reloadEvent) async {
+    // Generally this has the value 'success'; we update our data in any case.
+    // ignore: unused_local_variable
+    final status = reloadEvent.status;
+
+    // Refresh the list of scripts.
+    final scriptRefs = await _retrieveSortScripts(isolateRef);
+    for (var scriptRef in scriptRefs) {
+      _uriToScriptMap[scriptRef.uri] = scriptRef;
+    }
+
+    final removedScripts =
+        Set.of(_sortedScripts.value).difference(Set.of(scriptRefs));
+    final addedScripts =
+        Set.of(scriptRefs).difference(Set.of(_sortedScripts.value));
```

Comment:
I don't have another suggestion, but this seems like it could be expensive for apps with a large number of scripts

Reply:
I think the use of `Set`s will make this reasonably performant (`O(n)`), but agree that it could be improved. I filed https://github.com/dart-lang/sdk/issues/41668 to ask the VM for more explicit data about what scripts were added and removed.